### DISCUSSION TITLE
Can't activate licence when using Code With Me

### DISCUSSION THREAD

**Discussion by Sacro** [izhangzhihao/intellij-rainbow-brackets#2656] [Q&A] [answered]
Your issue may already be reported!
Please search on the [issues](https://github.com/izhangzhihao/intellij-rainbow-brackets/issues) and the [document](https://github.com/izhangzhihao/intellij-rainbow-brackets/blob/2020.3/README.md) and [discussions](https://github.com/izhangzhihao/intellij-rainbow-brackets/discussions) before creating one.

## Please check

- [ ] I already support this project.
- [X] I already paid for this plugin.
- [ ] I'm on a trial of this plugin.
- [ ] This issue/feature request is not reported before.

## Expected Behavior
* It should activate the licence when I'm coding remotely from my laptop to my desktop

## Current Behavior
* It tells me `The IDE licensing functionality is not complete initialization.`

## Possible Solution
Not sure what to try, are there any logs?

## Code snippet for reproduce (for bugs)
All I did was load a Goland project on my desktop, copy the code with me link and join it on my laptop

## Your Environment

* 2023.3.3
* Host - 
```
GoLand 2023.2
Build #GO-232.8660.185, built on July 26, 2023
Licensed to **REDACTED**
Subscription is active until October 18, 2023.
Runtime version: 17.0.7+7-b1000.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 32
Registry:
    suggest.all.run.configurations.from.context=TRUE
    debugger.new.tool.window.layout=true
    ide.completion.variant.limit=500
    ide.experimental.ui=true

Non-Bundled Plugins:
    Key Promoter X (2023.2.0)
    org.intellij.plugins.hcl (232.8660.88)
    com.jetbrains.ide.streamdeck (2023.2.0)
    org.toml.lang (232.8660.88)
    monokai-pro (1.10)
    com.intellij.tasks (232.8660.88)
    com.github.catppuccin.jetbrains (2.2.0)
    mobi.hsz.idea.gitignore (4.5.1)
    com.daniel-espinoza.inline-error (0.4.0)
    io.aesy.yamllint (0.3)
    com.intellij.kubernetes (232.8660.158)
    ru.adelf.idea.dotenv (2023.2)
    com.ypwang.plugin.go-linter (1.6.6)
    com.google.gct.core (23.8.1-api-version-222)
    org.sonarlint.idea (8.5.1.75093)
    izhangzhihao.rainbow.brackets (2023.3.3)
    intellij.prettierJS (232.8660.149)
```
* Client
```
JetBrains Client 2023.2
Build #JBC-232.8660.185, built on July 26, 2023
Runtime version: 17.0.7+7-b1000.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.4.10-zen2-1-zen
Guest in Code With Me (GO), direct control type: None
GC: G1 Young Generation, G1 Old Generation
Memory: 1500M
Cores: 16
Registry:
    debugger.new.tool.window.layout=true
    ide.experimental.ui=true
    rdclient.backend.actionHandlers=false

Non-Bundled Plugins:
    org.intellij.plugins.hcl (232.8660.88)
    izhangzhihao.rainbow.brackets (2023.3.3)
    com.intellij.plugins.colorful.darcula.colorscheme (232.8660.88)
    com.github.catppuccin.jetbrains (2.2.0)
    com.intellij.plugins.vscodekeymap (232.8660.88)
    com.intellij.plugins.monokai.colorscheme (232.8660.88)
    monokai-pro (1.10)
    Key Promoter X (2023.2.0)

Current Desktop: GNOME
```

**Comment by izhangzhihao** [comment, marked as answer]
Hi, you don't need to install Rainbow Brackets in the Code With Me **client** side, it works once the **server** side has this plugin.

<img width="1221" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/a734b0b2-746f-4ffd-b64a-9f89981a7bb1">

**Comment by izhangzhihao** [comment]
related to #2408 #2464

**Comment by izhangzhihao** [comment]
I will convert this issue to a discussion, and I will keep this open so it may help others.
If you have more questions, please discuss them there.

**Comment by Scarjit** [comment]
@izhangzhihao While the plugin itself seems to work with JB Gateway host, it fails to load the premium features.

Is there any workaround for that ?

![jetbrains_client64_HIciXymbmO](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/6317420/6d56a786-d7a8-4c18-af4d-5345ce5d8cb5)

Installing it on the client seems to have no effect at all.

**Reply by izhangzhihao** [reply]
Please check this issue for why gateway does not supported: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2584#issuecomment-1399424131

**Reply by izhangzhihao** [reply]
Just did some research, and I found that the JetBrains Client/Gateway does not support paid plugins at all: https://youtrack.jetbrains.com/issue/GTW-2304/JetBrains-Client-does-not-allow-me-to-use-paid-plugin

**Comment by izhangzhihao** [comment]
I'm consider to make this plugin incompatible with JetBrains Client. any thoughts? pls leave a comment here: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2708#issuecomment-1940685388

**Comment by izhangzhihao** [comment]
The text you provided is already in English, so no translation is needed. Here's the content with the format preserved:

Closed since the [Code With Me will sunset](https://blog.jetbrains.com/platform/2026/03/sunsetting-code-with-me/).
